### PR TITLE
runtime-rs: ch: Implement minimal implementation for missing thread/pid APIs

### PR DESCRIPTION
**Pull request by jodh-intel**
Add implementations for the following `Hypervisor` trait methods which simply return the same details as the `get_vmm_master_tid()` method:

- `get_thread_ids()`
- `get_pids()`

Fixes: #6438.

### CODE REVIEW

File: src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs [hunk 1]
```diff
@@ -676,7 +677,16 @@ impl CloudHypervisorInner {
     }
 
     pub(crate) async fn get_thread_ids(&self) -> Result<VcpuThreadIds> {
-        Ok(VcpuThreadIds::default())
+        let mut vcpus = HashMap::new();
+
+        let vcpu = 0;
```
Comment (cmaf) [inline src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs:682]: @jodh-intel is this always 0?

Comment (jodh-intel) [reply src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs:682]: @cmaf - well, I'm not sure what else we could do here. Thoughts?

File: src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs [hunk 2]
```diff
@@ -688,7 +698,9 @@ impl CloudHypervisorInner {
     }
 
     pub(crate) async fn get_pids(&self) -> Result<Vec<u32>> {
-        Ok(Vec::<u32>::new())
+        let pid = self.get_vmm_master_tid().await?;
```
Comment (amshinde) [inline src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs:701]: @jodh-intel We should probably return the virtio-fs deamon pid here as well.

Comment (amshinde) [reply src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs:701]: I just checked the golang implementation for the `GetPids` API. 
Its interesting that the qemu implementation returns both hypervisor+virtiofs pids, whereas clh just returns the hypervisor pid.
Point to note is that golang has a separate API for the virtiofs deamon:
https://github.com/kata-containers/kata-containers/blob/main/src/runtime/virtcontainers/hypervisor.go#L1108

This is missing from current rust hypervisor plugin API. It would not apply to dragonball anyway as it does not run a separate virtio-fs deamon.

Also, dragonball returns all the child pids in its implementation:
https://github.com/kata-containers/kata-containers/blob/main/src/runtime-rs/crates/hypervisor/src/dragonball/inner_hypervisor.rs#L110

This should also include virtio-fs thread id.

So in conclusion, I feel that with runtime-rs+clh, we should return hypervisor_pid + virtiofs_pid as well.

FYI, the only place I see the golang GetPids api being called is for metrics:
https://github.com/kata-containers/kata-containers/blob/main/src/runtime/virtcontainers/sandbox_metrics.go#L135

File: src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs (hunk 1 above)
Comment (amshinde) [reply src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs:682]: @jodh-intel This api is meant to return all the tids corresponding to all the vcpus that the VM uses. 

qemu has an qmp api `query-cpus-fast` that returns the details about the vcpus: https://www.qemu.org/docs/master/system/cpu-hotplug.html

I am not sure if such an api exists or has been added to clh since the go implementation. 
cc @likebreath  Can you let us know about this?

Else we may have to follow the same method that we used for the go-implementation i.e walk through the /proc/pid  tree:
https://github.com/kata-containers/kata-containers/blob/main/src/runtime/virtcontainers/clh.go#L760

Comment (likebreath) [reply src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs:682]: > qemu has an qmp api `query-cpus-fast` that returns the details about the vcpus: https://www.qemu.org/docs/master/system/cpu-hotplug.html
> 
> I am not sure if such an api exists or has been added to clh since the go implementation. cc @likebreath Can you let us know about this?

@amshinde is right. Cloud Hypervisor does not provide such pid information at this time. Of course, please feel free to send feature requests to the CLH community.

One clarification questions: what kind of PID information does kata want from VMM?

File: src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs (hunk 2 above)
Comment (jodh-intel) [reply src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs:701]: Hi @amshinde - I'm not sure that's needed since with the golang runtime, the CH driver spawns the virtiofsd but with runtime-rs, the runtime itself does that, not the CH driver.

Comment (amshinde) [reply src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs:701]: Ok, it makes sense to not include virtio-fs then

**Comment by jodh-intel** [comment]
This PR is a minimal implementation. We can't return the full details since CH doesn't provide that information currently. I've raised https://github.com/cloud-hypervisor/cloud-hypervisor/issues/6103 and https://github.com/kata-containers/kata-containers/issues/8799 for the "phase 2" implementation :smile: 

/cc @amshinde, @cmaf

**Review by amshinde — APPROVED** [review]
@jodh-intel I am approving this, lets open a separate PR to handle multiple VCPUs in a separate PR similar to the golang implementation.

**Review by cmaf — APPROVED** [review]
thanks @jodh-intel
